web/src/toga_web/widgets/dateinput.py

```diff
@@ -0,0 +1,50 @@
+import datetime
+
+from toga_web.libs import create_proxy
+
+from .base import Widget
+
+
+def py_date(native_date):
+    return datetime.datetime.strptime(native_date, "%Y-%m-%d").date()
+
+
+def native_date(py_date):
+    return py_date.strftime("%Y-%m-%d")
+
+
+class DateInput(Widget):
+    def create(self):
+        self._return_listener = None
+        self.native = self._create_native_widget("sl-input")
+        self.native.setAttribute("type", "date")
+
+        self.set_value(datetime.date.today().strftime("%Y-%m-%d"))
+        self.native.addEventListener("sl-change", create_proxy(self.on_change))
+
+    def get_value(self):
+        return py_date(self.native.value)
+
+    def set_value(self, value):
+        if value is None:
+            self.native.value = ""
+        self.native.value = value
```

Comment:
This doesn't appear to be fully working with initial values. In my testing, on Safari, the current date is being displayed on every date widget in the example app; on Chrome (on macOS), I get dd/mm/yyyy as the default values.

The "with max" and "with min and max" date examples should have an initial date of 2021-04-02; "any" and "with min" should have no initial value (although if the browser defaults to the current date as a browser behavior, that's fine).

Reply:
Not sure if this is impacting the behavior difference - but I'm using AU date formatting order, so I see "6/11/2023" as todays' date (6 Nov 2023). I'm not sure the extent to which browsers will be rejecting the default value of "%Y-%m-%d" on the basis of localised validation.

Reply:
I only tested this on chrome, I'll test this on firefox as well to see the differences (my apologies).

> The "with max" and "with min and max" date examples should have an initial date of 2021-04-02; "any" and "with min"

 but I'll definitely change this piece to have this functionality.